orangewidget/widget.py

```diff
@@ -111,9 +111,9 @@ class WidgetMetaClass(type(QDialog)):
     # pylint: disable=bad-classmethod-argument
     def __new__(mcs, name, bases, namespace, openclass=False, **kwargs):
         cls = super().__new__(mcs, name, bases, namespace, **kwargs)
+        cls.convert_signals()
```

Comment:
`convert_signals` is now also called for base classes for widgets, which are themselves not widgets.

This makes assignment of id's cleaner because it does not use MRO and checks each class just once.

A side-effect is that base class must define handlers for input signals; before, it suffice that derived class (an actual widget) did so. If this is a problem, I'd put assignment of ids out of `convert_signals` and call it separately.